### ISSUE TITLE
[BUG] Hypothesis filtering issue

### ISSUE THREAD

**Issue by magnustt** [schemathesis/schemathesis#1189]
When using Schemathesis 3.7.3 and above I get the error: 
"hypothesis.errors.FailedHealthCheck: It looks like your strategy is filtering out a lot of data. Health check found 50 filtered examples but only 0 good ones."

I assume it is connected to the filtering changes.

The API that is giving me trouble is one that takes the following parameters:

```yaml
{
openapi: 3.0.0
parameters:
  - schema:
      type: string
      pattern: '^[A-F0-9]{12}$'
      example: AB12CD34EF56
      minLength: 12
      maxLength: 12
    in: header
    name: Authorization
    description: authentication key
    required: true
}
```

**Environment (please complete the following information):**
Runs in a docker container during a Github workflow.

**Comment by Stranger6667** [comment]
Thanks for reporting!

I will take a look later today :)

**Comment by Stranger6667** [comment]
I overlooked that case - the "format" keyword injected into the final schema interferes badly with other keywords. I disabled this optimization for such cases. The fix will be released in `3.7.6` (I released an incomplete fix in `3.7.5`)

**Comment by magnustt** [comment]
Works in 3.7.7, thank you for a very fast fix! :)
